idaes/core/util/tables.py

```diff
@@ -91,7 +91,17 @@ def _stream_dict_add(sb, n, i=None):
     for n in streams.keys():
         if isinstance(streams[n], Arc):
             for i, a in streams[n].items():
-                sb = _get_state_from_port(a.ports[1], time_point)
+                try: 
+                    # if getting the StateBlock from the destination port
+                    # fails for any reason try the source port. This could
+                    # happen if a port does not have an associated 
+                    # StateBlock. For example a surrogate model may not
+                    # use state blocks, unit models may handle physical
+                    # properties without state blocks, or the port could
+                    # be used to serve the purpose of a translator block.
+                    sb = _get_state_from_port(a.ports[1], time_point)
+                except:
```

Comment:
Use of bare excepts is generally discouraged, but I won't insist on changing it.

Reply:
Yeah.  I know.  I wasn't sure of exactly all the ways this could fail, and if there is a mistake in the method the exception would get raised on the second try anyway.